### PR TITLE
[doc] Contributor fixes & new "finding reviewers" section

### PR DESCRIPTION
**Pull request by kallewoof**
I am summarising the suggestions given to me on IRC earlier regarding finding peer reviewers for pull requests.

I am also doing some general wording tweaks while at it.

I am somewhat concerned this is too wordy and maybe shouldn't be in `CONTRIBUTING.md` but in a separate file (`FINDINGPEERREVIEWERS.md` doesn't really float my boat). Feedback welcome.  This was one of the biggest frustrations I had when I started out, and I think it's worth some space *somewhere*.

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
Seems reasonable. Should definitely have such a section.

**Comment by MarcoFalke** [comment]
ACK

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -225,6 +225,36 @@ discussed extensively on the mailing list and IRC, be accompanied by a widely
...
+
+As most reviewers are themselves developers with their own projects, the review
+process can be quite lengthy, and some amount of patience is required. If you find
+that you've been waiting for a pull request to be given attention for several
+months, there may be a number of reasons for this, some of which you can do something
+about:
+
+  - It may be because of a feature freeze due to an upcoming release. During this time,
+    only bug reports are taken into consideration. If your pull request is a new feature,
+    it will be disregarded until the release is over. Wait for release.
```
Comment (laanwj) [inline CONTRIBUTING.md]: I think 'disregarded' is too strong here. It will be delayed, but there's no rule preventing e.g. discussion or people from testing it.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -225,6 +225,36 @@ discussed extensively on the mailing list and IRC, be accompanied by a widely
...
+### Finding Reviewers
+
+As most reviewers are themselves developers with their own projects, the review
+process can be quite lengthy, and some amount of patience is required. If you find
+that you've been waiting for a pull request to be given attention for several
+months, there may be a number of reasons for this, some of which you can do something
+about:
+
+  - It may be because of a feature freeze due to an upcoming release. During this time,
+    only bug reports are taken into consideration. If your pull request is a new feature,
```
Comment (laanwj) [inline CONTRIBUTING.md]: s/bug reports/bug fixes ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -225,6 +225,36 @@ discussed extensively on the mailing list and IRC, be accompanied by a widely
...
+    contribution, thundering silence is a good sign of widespread (mild) dislike of a given change
+    (because people don't assume *others* won't actually like the proposal). Don't take
+    that personally, though! Instead, take another critical look at what you are suggesting
+    and see if it: changes too much, is too broad, doesn't adhere to the [developer notes](doc/developer-notes.md),
+    is dangerous or insecure, is messily written, etc. Identify and address any of the
+    issues you find. Then ask e.g. on IRC if someone could give their opinion on the concept
+    itself.
+  - It may be because your code is too complex for all but a few people. And those people
+    may not have realized your pull request even exists. A great way to find people who
+    are qualified and care about the code you are touching is the [Git Blame feature](https://help.github.com/articles/tracing-changes-in-a-file/). Simply
```
Comment (laanwj) [inline CONTRIBUTING.md]: Good suggestion.

It looks like github also provides reviewer suggestions based on who touched the code before (see e.g. https://github.com/bitcoin/bitcoin/pull/10202, or any pull that hasn't got any reviewers yet)

**Comment by laanwj** [comment]
Looks good to me

File: CONTRIBUTING.md (hunk 1 above)
Comment (kallewoof) [inline CONTRIBUTING.md]: Makes sense. Changing to "will not be prioritized".

File: CONTRIBUTING.md (hunk 3 above)
Comment (kallewoof) [inline CONTRIBUTING.md]: That may be for members only. I don't see any suggestions on that PR but I've seen suggestions on PR's on repos to which I have merge access.

**Comment by kallewoof** [comment]
Addressed issues pointed out by @laanwj. If anyone has any other improvements they'd like to see do let me know!

**Comment by paveljanik** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/10211/commits/3ddd227c2f248476ce1689feb98f2532ea95ec91

Nice addition, thank you!
